core/src/main/java/org/apache/iceberg/BaseDistributedDataScan.java

```diff
@@ -223,7 +223,18 @@ private List<ManifestFile> filterManifests(List<ManifestFile> manifests) {
         .collect(Collectors.toList());
   }
 
-  protected boolean shouldPlanLocally(PlanningMode mode, List<ManifestFile> manifests) {
```

Comment:
I initially made this protected but I re-considered to make it private for now. Making this method protected will be make future changes harder because the compatibility must be preserved. We know this class will be evolving.